src/stdx.zig

```diff
@@ -428,3 +428,65 @@ pub fn update(base: anytype, diff: anytype) @TypeOf(base) {
     }
     return updated;
 }
+
+// TODO(Zig): No need for this function once Zig is upgraded
+// and @fieldParentPtr() can be used for unions.
+// See: https://github.com/ziglang/zig/issues/6611.
+
+pub fn UnionFieldParentPtrType(
+    comptime Union: type,
+    comptime field: std.meta.FieldEnum(Union),
+    comptime Child: type,
+) type {
+    assert(@typeInfo(Union) == .Union);
+    const Field = std.meta.fieldInfo(Union, field).field_type;
+
+    assert(std.meta.trait.isSingleItemPtr(Child));
+    assert(std.meta.Child(Child) == Field);
+
+    return if (std.meta.trait.isConstPtr(Child))
+        *const Union
+    else
+        *Union;
+}
```

Comment:
Could this be omitted in favor of assuming `*Union` for `union_field_parent_ptr` usages?

Reply:
Yes, we can do that, the previous implementation removed the *const qualifier by doing intToPtr, so I wanted to fix it and keep the stdx implementation generic for future use. But anyway, it’s not our use case.